postpic/datahandling.py

```diff
@@ -222,12 +228,31 @@ def __init__(self, matrix, xedges=None, yedges=None, zedges=None, name='', unit=
         # self.axes_transform_state is False for axes which live in spatial domain
         # and it is True for axes which live in frequency domain
         # This assumes that fields are initially created in spatial domain.
-        self.axes_transform_state = [False] * len(self.matrix.shape)
+        self.axes_transform_state = [False] * len(self.shape)
 
         # self.transformed_axes_origins stores the starting values of the grid
         # from before the last transform was executed, this is used to
         # recreate the correct axis interval upon inverse transform
-        self.transformed_axes_origins = [None] * len(self.matrix.shape)
+        self.transformed_axes_origins = [None] * len(self.shape)
+
+    def __copy__(self):
+        '''
+        returns a shallow copy of the object.
+        This method is called by `copy.copy(obj)`.
+        Just copy enough to create copies for operator overloading.
+        For example:
+        axes objects are not copied, but the list holding the references to
+        the axis objects is.
```

Comment:
I think `axes` objects should be copied as well. Axis has `grid.setter` und `setextent` which might still be used by a user without taking care to copy the object first. Otherwise this caveat has to be documented in a prominent place.

Reply:
docstring only